Code/Framework/AzFramework/AzFramework/DocumentPropertyEditor/ReflectionAdapter.cpp

```diff
@@ -30,27 +30,128 @@ namespace AZ::DocumentPropertyEditor
 
         struct BoundContainer
         {
-            AZ::SerializeContext::IDataContainer* m_container;
-            void* m_instance;
-            void* m_elementInstance = nullptr;
-            size_t m_elementIndex = 0;
+            // For constructing non-nested containers
+            BoundContainer(AZ::SerializeContext::IDataContainer* container, void* containerInstance)
+                : m_container(container)
+                , m_containerInstance(containerInstance)
+            {
+            }
+
+            // For constructing non-container elements
+            BoundContainer(
+                AZ::SerializeContext::IDataContainer* container,
+                void* containerInstance,
+                void* elementInstance)
+                : m_parentContainer(container)
+                , m_parentContainerInstance(containerInstance)
+                , m_elementInstance(elementInstance)
+            {
+            }
+
+            // For constructing nested containers
+            BoundContainer(
+                AZ::SerializeContext::IDataContainer* parentContainer,
+                void* parentContainerInstance,
+                AZ::SerializeContext::IDataContainer* container,
+                void* containerInstance,
+                void* overrideInstance)
+                : m_parentContainer(parentContainer)
+                , m_parentContainerInstance(parentContainerInstance)
+                , m_container(container)
+                , m_containerInstance(containerInstance)
+                , m_elementOverrideInstance(overrideInstance)
+            {
+            }
```

Comment:
It seems like these functions should be private, as it seems that you intend that Bound Containers be only created publically through `CreateBoundContainer` correct?

Reply:
Excellent point. I was thinking there might be some case where a BoundContainer could be created otherwise, but better to make these private. This whole thing is starting to read a lot more like a class instead of a struct, semantically.